msbuild/Xamarin.MacDev.Tasks/Tasks/XcodeBuild.cs

```diff
@@ -0,0 +1,106 @@
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Xamarin.MacDev.Tasks {
+	public class XcodeBuild : XamarinToolTaskWithOutput {
+		public override string TaskPrefix => "XCBD";
+
+		protected override string ToolName => "xcodebuild";
+
+		public string Command { get; set; } = string.Empty;
+
+		public string Arguments { get; set; } = string.Empty;
+
+
+		// archive arguments
+		public string ProjectPath { get; set; } = string.Empty;
+
+		public string SchemeName { get; set; } = string.Empty;
+
+		public string Configuration { get; set; } = string.Empty;
+
+		public string ArchivePlatform { get; set; } = string.Empty;
+
+		public string ArchiveOutputPath { get; set; } = string.Empty;
+
+		public string DerivedDataPath { get; set; } = string.Empty;
+
+		public string PackageCachePath { get; set; } = string.Empty;
+
+		// create-xcframework arguments
+		public ITaskItem [] FrameworkArchives { get; set; } = Array.Empty<ITaskItem> ();
+
+		public string FrameworkName { get; set; } = string.Empty;
+
+		public string FrameworkOutputPath { get; set; } = string.Empty;
+
+
+		protected override string GenerateFullPathToTool ()
+		{
+			return Path.Combine ("/usr", "bin", ToolExe);
+		}
+
+		protected override string GenerateCommandLineCommands ()
+		{
+			var cmd = new CommandLineBuilder ();
+
+			if (!string.IsNullOrEmpty (Command))
+				cmd.AppendSwitch (Command);
+
+			if (!string.IsNullOrEmpty (ProjectPath))
+				cmd.AppendSwitchIfNotNull ("-project ", ProjectPath);
+
+			if (!string.IsNullOrEmpty (SchemeName))
+				cmd.AppendSwitchIfNotNull ("-scheme ", SchemeName);
+
+			if (!string.IsNullOrEmpty (Configuration))
+				cmd.AppendSwitchIfNotNull ("-configuration ", Configuration);
+
+			if (!string.IsNullOrEmpty (ArchivePlatform))
+				cmd.AppendSwitchIfNotNull ("-destination ", ArchivePlatform);
+
+			if (!string.IsNullOrEmpty (ArchiveOutputPath))
+				cmd.AppendSwitchIfNotNull ("-archivePath ", ArchiveOutputPath);
+
+			if (!string.IsNullOrEmpty (DerivedDataPath))
+				cmd.AppendSwitchIfNotNull ("-derivedDataPath ", DerivedDataPath);
+
+			if (!string.IsNullOrEmpty (PackageCachePath))
+				cmd.AppendSwitchIfNotNull ("-packageCachePath ", PackageCachePath);
+
+			foreach (var frameworkArchive in FrameworkArchives) {
+				cmd.AppendSwitchIfNotNull ("-archive ", frameworkArchive.ItemSpec);
+				cmd.AppendSwitchIfNotNull ("-framework ", FrameworkName);
+			}
+
+			if (!string.IsNullOrEmpty (FrameworkOutputPath))
+				cmd.AppendSwitchIfNotNull ("-output ", FrameworkOutputPath);
+
+			if (!string.IsNullOrEmpty (Arguments))
+				cmd.AppendSwitch (Arguments);
+
+			return cmd.ToString ();
+		}
+
+		public override bool RunTask ()
+		{
+			if (RuntimeInformation.IsOSPlatform (OSPlatform.OSX)) {
```

Comment:
@rolfbjarne @emaf @mauroa is there a preferred way to ensure that certain build tasks only execute on macOS? It may be worth discussing remote mac build support for this in the future, but for now these new tasks are only intended to work on macOS.

Reply:
I was thinking it would be helpful to provide a warning or message to acknowledge that the project declares an Xcode project but that it won't be built when building such a project on Windows.

I can move that to the targets instead, or remove it entirely if we'd rather just skip these targets silently there?

Reply:
People don't read warnings 😉 You could also just show a warning in the targets as well, although it would complicate the logic a little bit.

Reply:
I removed the warnings, and put the import of the new targets file behind this condition:

```xml
<Import Project="Microsoft.MaciOS.Sdk.Xcode.targets" Condition=" $([MSBuild]::IsOSPlatform('osx')) " />
```